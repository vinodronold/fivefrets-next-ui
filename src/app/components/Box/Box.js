import React, { PureComponent } from 'react'
import { func, node, object } from 'prop-types'
import { color } from '../../constants/theme'

export const BoxStyle = `
display: flex;
align-items: center;
min-height: 3rem;
margin: 1rem auto;
background-color: ${color.bg()};
width: 80%;
`

export default class Box extends PureComponent {
  static propTypes = {
    children: node,
    style: object,
    onClick: func
  }

  render() {
    const { children, style, onClick } = this.props
    return (
      <div style={style} onClick={onClick}>
        {children}
        <style jsx>{`
          div {
            ${BoxStyle};
            justify-content: center;
          }
        `}</style>
      </div>
    )
  }
}
