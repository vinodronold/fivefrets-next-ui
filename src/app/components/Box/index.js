import Box, { BoxStyle } from './Box'
export default Box
export { BoxStyle }
