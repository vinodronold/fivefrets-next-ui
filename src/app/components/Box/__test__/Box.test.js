import React from 'react'
import renderer from 'react-test-renderer'
import Box from '../Box'

describe('Box', () => {
  it('should match the snapshot', () => {
    const _box = renderer.create(<Box>BOX</Box>)
    expect(_box).toMatchSnapshot()
  })
})
