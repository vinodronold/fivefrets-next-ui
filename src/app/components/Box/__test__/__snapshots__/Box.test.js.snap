// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should match the snapshot 1`] = `
<div
  className="jsx-1762149615"
  onClick={undefined}
  style={undefined}
>
  BOX
</div>
`;
